### DISCUSSION TITLE
Grove - LCD RGB Backlight | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#96] [Q&A] [answered]
# Grove - LCD RGB Backlight | Seeed Studio Wiki

Grove - LCD RGB Backlight

https://wiki.seeedstudio.com/Grove-LCD_RGB_Backlight/

**Comment by ffsianhall** [comment]
There is mention tht the display has 4 addresses but uses 2, can the address be changed?

**Reply by MatthewJeffson** [reply]
Hi, this can be an interesting thought. Would you mind help to test it? We will be so much greatful:D

**Reply by ffsianhall** [reply]
i would love some help to test it, i would like to know the address registers.

**Reply by MatthewJeffson** [reply]
You got me with this one XD I will try my best :P

**Comment by CleoQc** [comment]
I am trying to figure out what changed with V5. 
I have a V2 with me and just got a V5. 
Using the arduino example sketch, the V2 will change the background colour, but not the text.
The V5 will display the text but not the background colour.

If I use a GrovePi+, the V2 works flawlessly, both RGB and text, and the V5 shows text but the background colour doesn't work properly. By that I mean I get the wrong colours (mostly various shades of white) and it flashes like crazy.

Yes, V5 has had an I2C address change and I adjusted the python code accordingly. 
0x3e and 0x70 for V2
0x3e and 0x30 for V5

I have a friend with a V4, and they report 0x62 and 0x70, no 0x3e for them. 

Help?

**Reply by benhaotang** [reply]
I also have this problem, I have V4, no 0x3e, only RGB works. there is 0x62 for RGB and works fine, but 0x70 don't work at all.

**Reply by FrancoisTrocque** [reply, marked as answer]
Hello I've got the same problem and I solves the issue
In the Cpp file the "_initialized = 1;" line in "void rgb_lcd::begin" function is placed at the end of the function.
So all previous "command" function don't work because they test "_initialized" before doing the job.
To Fix place the "_initialized = 1;"  line at the begining of the function 
i.e. After "_wire->begin();" line 75
and all become OK
This is a clue that the lib was provided without Testing it (:-(

**Reply by limengdu** [reply]
Thank you for your feedback, we have fixed this issue.
